### ISSUE TITLE
Add stylelint-disable-no-empty rule

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1814]
Disable empty `stylelint-disable` in files.
Invalid:

``` css
/* stylelint-disable */
.selector {
  color: red;
}
```

``` css
.selector {
  color: red; // stylelint-disable
}
```

Valid:

``` css
/* stylelint-disable color-named */
.selector {
  color: red;
}
```

``` css
.selector {
  color: red; // stylelint-disable color-named
}
```

Some people in the team think that turning off `stylelint`, review will be faster accepted. Be good if CI down for this bad guys :smiling_imp:

**Comment by jeddy3** [comment]
`stylelint-disable-no-empty` sgtm as a core rule.

`stylelint-disable-no-abusive` as a plugin as `abusive` is [ambiguous](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#creating-new-rules).

**Comment by evilebottnawi** [comment]
@jeddy3 right, `stylelint-disable-no-empty` sgtm

**Comment by jeddy3** [comment]
- Name: `stylelint-disable-no-empty`
- Description: "Disallow empty `stylelint-disable` comments"
- Extended description: "When `stylelint-disable` comments are empty they disable *all* the rules in the disable range. This rule encourages disabling *individual* rules."
- Primary options: `true`
- Secondary options: none
- Message: `Unexpected empty `stylelint-disable` comment`
- Section: "`stylelint-disable` comment"

**Comment by jeddy3** [comment]
Superseded by #2292 and #2291

**Comment by Avcajaraville** [comment]
Being looking for a way to disable this rule for a specific folder inside my project (in particular, we generate angular components via cli, and it is fine [for us] to commit empty files).

The only way I manage to do this was to extend .stylelintrc and add a particular rule there:

```json
{
  "extends": ["../.stylelintrc"],
  "rules": {
    "no-empty-source": null
  }
}
```

I found this extremely odd, reasons being:
1. Passing null as an option, why not adding support for `false` ?
2. There is no way of adding a pattern to be ignored at rule level.

What I mean with point 2, is to have some sort of configuration like:

```json
...
"rules": {
  {
    "files": ["*.component.scss"],
    "rules": {
      "no-empty-source": false
    }
  }
}
...
```

This is how `eslint` handles it.


Sorry for resurrecting an old issue.

**Comment by jeddy3** [comment]
@Avcajaraville Thanks for the comment.

Yes, extending your config and running stylelint twice is the workaround until the [`overrides` property is added](https://github.com/stylelint/stylelint/issues/3128). That issue is waiting for someone to champion it.

---

For anyone else revisiting this thread, broad stylelint commands can be disallowed using the new [`report*` configuration options](https://stylelint.io/user-guide/configure#disable-errors).
